### PR TITLE
[v1.1.0] Feat: Admin Access Logs

### PR DESCRIPTION
**Pull request by DanStough**
This adds access logs to Envoy's administration interface. This is controlled by the `proxy-defaults` configuration entry, which gets passed through the `GetBootstrapParamaters` RPC. RPC changes on the Consul side are already merged: https://github.com/hashicorp/consul/pull/15951.

Application Access Logs are handled dynamically through xDS.

This PR also adds integration tests on each PR. More work is needed there to keep the image tag fresh for Consul.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -6,7 +6,7 @@ require (
 	github.com/adamthesax/grpc-proxy v0.0.0-20220525203857-13e92d14f87a
 	github.com/armon/go-metrics v0.4.1
 	github.com/hashicorp/consul-server-connection-manager v0.1.0
-	github.com/hashicorp/consul/proto-public v0.1.1
+	github.com/hashicorp/consul/proto-public v0.1.2-0.20230111184009-6d2880e89478
```
Comment (DanStough) [inline go.mod:9]: Not sure what we want to do here:
1. Merge and change this as part of the release
1. Leave it unmerged and fix right after the Consul Core release.

File: integration-tests/main_test.go [hunk 2]
```diff
@@ -59,22 +67,23 @@ func TestMain(m *testing.M) {
...
+//   - Creating proxy-defaults to set the default protocol to HTTP and prometheus
+//     bind address. Also set access logs on the admin interface of Envoy
+//   - Creating an L7/HTTP intention to allow "frontend" to talk to "backend".
+//   - Making an HTTP request through the "frontend" sidecar's exposed "backend"
+//     port.
+//   - Setting the intention action to deny.
+//   - Attempting to make the same request and checking that it fails.
+//   - Making DNS queries against the frontend dataplane's UDP and TCP DNS proxies.
+//   - Scraping the prometheus merged metrics endpoint.
+//   - Make a call to Envoy's admin interface and check for the access logs.
```
Comment (DanStough) [inline integration-tests/main_test.go:86]: Pretty sure this is a go 1.19 formatting thing.

**Comment by david-yu** [comment]
We probably need to ensure the this only gets released with the 1.1.x release branch, not the upcoming 1.0.x patch release.

**Review by thisisnotashwin — APPROVED** [review]
Nice job on the integration test and getting that threaded through into CI!! This looks great!!

File: go.mod (hunk 1 above)
Comment (DanStough) [reply go.mod:9]: Made a PR to update the release process in our docs IAW option 1. 👍

**Pull request merged**